### PR TITLE
Xw 311

### PR DESCRIPTION
**Pull request by bognix**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/VKTests.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import com.wikia.webdriver.common.core.annotations.Execute;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
 import com.wikia.webdriver.pageobjectsfactory.componentobject.mercury.NavigationSideComponentObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.widget.TwitterWidgetPageObject;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/VKTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.widget.TwitterWidgetPageObject'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WeiboTests.java [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import com.wikia.webdriver.common.core.annotations.Execute;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
 import com.wikia.webdriver.pageobjectsfactory.componentobject.mercury.NavigationSideComponentObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.widget.VKWidgetPageObject;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/WeiboTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.widget.VKWidgetPageObject'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/SpotifyTests.java [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import com.wikia.webdriver.pageobjectsfactory.componentobject.mercury.NavigationSideComponentObject;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.widget.SpotifyWidgetPageObject;
 
+import com.wikia.webdriver.pageobjectsfactory.pageobject.widget.TwitterWidgetPageObject;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/SpotifyTests.java]: MINOR Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.widget.TwitterWidgetPageObject'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/SoundCloudWidgetPageObject.java [hunk 4]
```diff
@@ -3,6 +3,7 @@
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/SoundCloudWidgetPageObject.java]: MAJOR Replace this usage of Sun classes by ones from the Java API. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1191)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 6 major
* MINOR 3 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/2238a56a95d5ba119942429df74d6f78d9a2c0e5/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/PollsnackWidgetPageObject.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/2238a56a95d5ba119942429df74d6f78d9a2c0e5/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/SpotifyWidgetPageObject.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/2238a56a95d5ba119942429df74d6f78d9a2c0e5/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/TwitterWidgetPageObject.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/2238a56a95d5ba119942429df74d6f78d9a2c0e5/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/VKWidgetPageObject.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)
* MAJOR [1 duplicated blocks of code must be removed.](https://github.com/Wikia/selenium-tests/blob/2238a56a95d5ba119942429df74d6f78d9a2c0e5/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/WeiboWidgetPageObject.java) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=common-java%3ADuplicatedBlocks)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/SpotifyTests.java [hunk 5]
```diff
@@ -24,6 +26,9 @@
   @BeforeMethod(alwaysRun = true)
   public void prepareTest() {
     driver.manage().timeouts().pageLoadTimeout(100, TimeUnit.SECONDS);
+    //@TODO XW-314 - Login is necessary to bypass cache
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/SpotifyTests.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/TwitterWidgetPageObject.java [hunk 6]
```diff
@@ -18,6 +18,9 @@
   private static final String ARTICLE_NAME = "TwitterWidget";
   private static final String TAG =
       "<twitter widget-id=\"522824386202447873\" screen-name=\"sfbart\" />";
+  private static final String INCORRECT_TAG =
+      "<twitter />";
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/TwitterWidgetPageObject.java]: use one line please

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/VKWidgetPageObject.java [hunk 7]
```diff
@@ -18,6 +18,9 @@
   private static final String ARTICLE_NAME = "VKWidget";
   private static final String TAG =
       "<vk group-id=\"12345\" />";
+  private static final String INCORRECT_TAG =
+      "<vk />";
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/VKWidgetPageObject.java]: use one line please

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/VKWidgetPageObject.java [hunk 8]
```diff
@@ -35,6 +38,15 @@ protected String getTag() {
     return TAG;
   }
 
+
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/VKWidgetPageObject.java]: remove redundant line

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/PollsnackTests.java [hunk 9]
```diff
@@ -63,4 +63,14 @@ public void MercuryPollsnackWidgetTest_003_isLoadedOnSecondVisitFromDifferentArt
...
+
+  @Test(groups = "MercuryPollsnackWidgetTest_006")
+  @Execute(onWikia = "mercuryautomationtesting")
+  public void MercuryPollsnackWidgetTest_006_isErrorPresent() {
+    PollsnackWidgetPageObject pollsnackWidget = new PollsnackWidgetPageObject(driver);
+
+    pollsnackWidget.createIncorrectAndNavigate(wikiURL);
+    Assertion.assertTrue(pollsnackWidget.isErrorPresent(), MercuryMessages.INVISIBLE_MSG);
+  }
+
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/PollsnackTests.java]: remove redundant line

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/SpotifyTests.java (hunk 5 above)
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/widgettests/SpotifyTests.java]: please add one free line above TODO for better readability

**Comment by tomasznapieralski** [comment]
LGTM

**Pull request merged**